### ISSUE TITLE
Running Jar On IKVM fails with System.TypeInitializationException

### ISSUE THREAD

**Issue by lemonlion** [ikvmnet/ikvm#466]
I am attempting to run PlantUml in on IKVM, but it seems to just fail with a System.TypeInitializationException

To aid anyone who want to try help fixing the issue I have created a public repository of the code https://github.com/lemonlion/PlantUmlIkvmTest

First I went to https://plantuml.com/download and downloaded the latest MIT licensed version (`plantuml-mit-1.2023.13.jar`).  I then created a new C# .Net 6 console project named `PlantUmlIkvm` and added a new folder called `PlantUml` and put the jar in there.

Next I installed all the packages mentioned in the [IKVM Readme](https://github.com/ikvmnet/ikvm/blob/main/README.md) and I also installed `IKVM.Maven.Sdk`

```
    <ItemGroup>
        <PackageReference Include="IKVM" Version="8.7.3" />
        <PackageReference Include="IKVM.Image" Version="8.7.3" />
        <PackageReference Include="IKVM.Image.JDK" Version="8.7.3" />
        <PackageReference Include="IKVM.Image.JRE" Version="8.7.3" />
        <PackageReference Include="IKVM.Maven.Sdk" Version="1.6.5" />
        <PackageReference Include="IKVM.MSBuild" Version="8.7.3" />
    </ItemGroup>
```

Next I added an IkvmReference tag in my project file as per the IKVM readme.  I also put Debug true to allow for any debugging functionality.

```
    <ItemGroup>
        <IkvmReference Include="PlantUml\plantuml-mit-1.2023.13.jar" >
            <Debug>true</Debug>
        </IkvmReference>
    </ItemGroup>
```

At this point the entire solution looks like this:

[![enter image description here](https://i.stack.imgur.com/bEG0d.png)](https://i.stack.imgur.com/bEG0d.png)

[![enter image description here](https://i.stack.imgur.com/AoeHo.png)](https://i.stack.imgur.com/AoeHo.png)

I then rebuilt the entire solution.  I look in `C:/Code/PlantUmlIkvm/bin/Debug/net6.0` and am pleased to see that a dll has been created called `net.sourceforge.plantuml.dll` indicating that IKVM has successfully translated the jar into something .net compatible.

[![enter image description here](https://i.stack.imgur.com/k8JdM.png)](https://i.stack.imgur.com/k8JdM.png)

I then attempt to use PlantUml from my solution.  Besides some intellisense weirdness (the IDE isn't always recognising the classes as classes), the new classes appear to be picked up by visual studio and I can decompile into them using Visual Studio/Resharper.  I follow the [PlantUml tutorial from their main page](https://plantuml.com/api) for creating a PNG image from some basic plantUml and so I enter the following code into the main Program.cs:

```
    using System.Reflection;
    using com.plantuml.api.cheerpj;
    using com.plantuml.api.cheerpj.v1;
    using java.io;
    using net.sourceforge.plantuml;
    using net.sourceforge.plantuml.api;
    using net.sourceforge.plantuml.security;
    using net.sourceforge.plantuml.sequencediagram;
    using sun.misc;
    using sun.net.www.content.image;

    namespace PlantUmlIkvm;

    public class Program
    {
        public static void Main(string[] args)
        {
            var uml = "@startuml\n";
            uml += "Bob -> Alice : hello\n";
            uml += "@enduml\n";
            var reader = new net.sourceforge.plantuml.SourceStringReader(uml);
            OutputStream png = new Base64OutputStream();
            var desc = reader.outputImage(png).getDescription();
        }
    }
```

The program builds successfully.  It runs successfully up to a point, but fails on this line:

`
    var desc = reader.outputImage(png).getDescription();
`

The line fails with message 

`System.TypeInitializationException: 'The type initializer for 'net.sourceforge.plantuml.FileFormat' threw an exception.'`

[![enter image description here](https://i.stack.imgur.com/zaDWD.png)](https://i.stack.imgur.com/zaDWD.png)

[![enter image description here](https://i.stack.imgur.com/WGXcY.png)](https://i.stack.imgur.com/WGXcY.png)

Outer Exception TargetSite:

[![enter image description here](https://i.stack.imgur.com/7zrgB.png)](https://i.stack.imgur.com/7zrgB.png)

Inner Exception TargetSite:

[![enter image description here][1]][1]

The main Stack Trace:

[![enter image description here](https://i.stack.imgur.com/RClXU.png)](https://i.stack.imgur.com/RClXU.png)

The inner exception Stack Trace:

[![enter image description here](https://i.stack.imgur.com/V2mN7.png)](https://i.stack.imgur.com/V2mN7.png)

I believe the error is happening when it tries to create a FileFormatOption, hence when I insert this additional line it fails there first:

```
var fileFormat = new FileFormatOption(FileFormat.PNG);
```

[![enter image description here](https://i.stack.imgur.com/z9RTQ.png)](https://i.stack.imgur.com/z9RTQ.png)

I am running all this on Windows 10.

I imagine this is more of an IKVM issue than a PlantUml issue, but open to any ideas.

**Comment by wasabii** [comment]
Looks like this goes into AWT, which we don't currently support.

**Comment by wasabii** [comment]
#144

**Comment by lemonlion** [comment]
@wasabii Ah ok - thanks for your reply.  Is there any workaround at all that you could suggest?

**Comment by wasabii** [comment]
Nope. Not as it stands currently. There's probably a possibility of getting headless mode working, but I'm just not sure what that requires right now.

**Comment by lemonlion** [comment]
Ah that's sad to hear - PlantUml is used by so many .net developers - being able to run it in .Net would be a godsend.

**Comment by lytico** [comment]
@lemonlion plantuml works not with IKVM 8.9.1 https://github.com/ikvmnet/ikvm/releases/tag/8.9.1
